src/middleware/router.coffee

```diff
@@ -166,6 +183,7 @@ buildNonPrimarySendRequestPromise = (ctx, route, options, path) ->
 
     ctx.routes = [] if not ctx.routes
     ctx.routes.push routeObj
+    exports.nonPrimaryRoutes.push routeObj
```

Comment:
I needed this to keep track of the async non primary routes for my tests. because once the response has been sent to the client, the context gets reset.